### ISSUE TITLE
Blazor Webassembly shows wrong time in UI

### ISSUE THREAD

**Issue by fingers10** [dotnet/runtime#45421]
I'm developing my app in `.Net 5.0 Blazor WebAssembly`. I'm displaying current time in HTML. But the time displayed is always `-5.30 Hrs` the time displayed in my system clock. My computer time is set to `UTC+5.30` time zone.

After some googling I found the following in [official docs](https://docs.microsoft.com/en-us/aspnet/core/blazor/globalization-localization?view=aspnetcore-5.0#blazor-webassembly).

I added the below tags in `.csproj` file but still no luck.

Here is my code:

**.csproj:**
```xml
<PropertyGroup>
    <TargetFramework>net5.0</TargetFramework>
    <BlazorWebAssemblyLoadAllGlobalizationData>true</BlazorWebAssemblyLoadAllGlobalizationData>
    <InvariantGlobalization>true</InvariantGlobalization>
    <ServiceWorkerAssetsManifest>service-worker-assets.js</ServiceWorkerAssetsManifest>
</PropertyGroup>

<ItemGroup>
    <PackageReference Include="Microsoft.AspNetCore.Components.WebAssembly" Version="5.0.0" />
    <PackageReference Include="Microsoft.AspNetCore.Components.WebAssembly.DevServer" Version="5.0.0" PrivateAssets="all" />
    <PackageReference Include="Microsoft.Extensions.Localization" Version="5.0.0" />
    <PackageReference Include="System.Net.Http.Json" Version="5.0.0" />
</ItemGroup>
```
**Code:**
```razor
@using System.Threading
@implements IDisposable

<p>@CurrentDateTime</p>

@code {
    private Timer _timer;
    public string CurrentDateTime { get; set; }

    protected override void OnAfterRender(bool firstRender)
    {
        if (firstRender)
        {
            _timer = new Timer(DateTimeCallback, null, TimeSpan.FromSeconds(1), TimeSpan.FromSeconds(1));
        }
    }

    private async void DateTimeCallback(object state)
    {
        CurrentDateTime = DateTimeOffset.UtcNow.ToLocalTime().ToString("dd MMM yyyy hh:mm:ss tt");
        await InvokeAsync(StateHasChanged);
    }

    public void Dispose()
    {
        _timer?.Dispose();
    }
}
```

**Time Screenshot:**

[![enter image description here][1]][1]

Please assist on what I'm missing.

  [1]: https://i.stack.imgur.com/3lWqN.png

**Comment by mkArtakMSFT** [comment]
@lewing FYI

**Comment by tqiu8** [comment]
@fingers10 What timezone are you working in? I'm trying to repro this

**Comment by pranavkm** [comment]
Specifically the result of printing `Intl.DateTimeFormat().resolvedOptions().timeZone` in your browser console would help.

**Comment by fingers10** [comment]
@tqiu8 my system time zone details.
![image](https://user-images.githubusercontent.com/43729469/100814204-34614d80-3467-11eb-9588-b13a1052c4c0.png)

@pranavkm here is the console log output
![image](https://user-images.githubusercontent.com/43729469/100814253-4ba03b00-3467-11eb-93e6-2673ce6ad9c2.png)

**Comment by lewing** [comment]
Asia/Calcutta appears to be one of the obsolete zone names that should have been fixed in https://github.com/dotnet/runtime/pull/43982 which will be included in the 5.0.1 Servicing release.  Without that fix the timezone would essentially always be set to UTC.

**Comment by iAmBipinPaul** [comment]
Hi , @lewing 
I noticed that it shows correct local time on `mozilla firefox`

**Comment by fingers10** [comment]
Just an update. After updating to .net 5.0.1 I can able to see correct time in my browser. Was this fixed?

**Comment by xperiandri** [comment]
I have the same issue.
`TimeZone.CurrentTimeZone` is always UTC

**Comment by xperiandri** [comment]
Checked .NET 5 - not fixed

**Comment by fingers10** [comment]
It's working after updating to `Microsoft.AspNetCore.Components.WebAssembly` and `Microsoft.AspNetCore.Components.WebAssembly.DevServer` version `5.0.1` from `5.0.0`

**Comment by lewing** [comment]
Closing this as fixed in 5.0.1
